onnxruntime/core/util/math_cpu.cc

```diff
@@ -306,44 +317,58 @@
     int64_t stride_h,
     int64_t stride_w,
     T* data_col,
+    ThreadPool* tp,
     T padding_value) {
   const int64_t output_h = (height + pad_b + pad_t - (dilation_h * (kernel_h - 1) + 1)) / stride_h + 1;
   const int64_t output_w = (width + pad_l + pad_r - (dilation_w * (kernel_w - 1) + 1)) / stride_w + 1;
 
+  const auto dop = ThreadPool::DegreeOfParallelism(tp);
+  constexpr int64_t cost_per_batch = 6144;
```

Comment:
>6144 [](http://example.com/codeflow?start=37&length=4)

How is it computed?

Reply:
This is 6 * 1024. I iterated in the range of 4k - 8K and 6K gave me the best batching performance. Even though the lambdas wrap individual channels, the batching computation is around the actual number of elements being processed, so this makes it shape independent and widely applicable and ensures that we process minimum elements per thread.
At the same time, we do not want to occupy all the threads available, only those needed.

Reply:
I feel cautious when seeing hard-code number. Will it vary for different hardware?